### PR TITLE
fix package path issues when packing content into non-default locations inside nupkg

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes: https://github.com/NuGet/Home/issues/4321

Added extensive test coverage for all possible input/output combinations of source path and packagepath.

CC: @rrelyea @emgarten @alpaix @mishra14 @jainaashish @nkolev92 @zhili1208

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs [hunk 1]
```diff
@@ -364,33 +364,16 @@ private void InitNuspecOutputPath(IPackTaskRequest<IMSBuildItem> request, PackAr
...
 
             var buildAction = BuildAction.Parse(packageFile.GetProperty("BuildAction"));
 
             // TODO: Do the work to get the right language of the project, tracked via https://github.com/NuGet/Home/issues/4100
             var language = buildAction.Equals(BuildAction.Compile) ? "cs" : "any";
 
+
             var setOfTargetPaths = new HashSet<string>(targetPaths, StringComparer.Ordinal);
-            if (setOfTargetPaths.Remove("contentFiles" + Path.DirectorySeparatorChar) 
+            if (setOfTargetPaths.Remove("contentFiles" + Path.DirectorySeparatorChar)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs:376]: does this need to check for the AltDirectorySeparatorChar also?

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs [hunk 2]
```diff
@@ -400,6 +383,49 @@ private void InitNuspecOutputPath(IPackTaskRequest<IMSBuildItem> request, PackAr
...
+                // If identity is not a rooted path, then it is a relative path to the project directory
+                else if (identity.EndsWith(Path.GetFileName(sourcePath), StringComparison.CurrentCultureIgnoreCase))
+                {
+                    identity = Path.GetDirectoryName(identity);
+                }
+
+                foreach (var targetPath in setOfTargetPaths)
+                {
+                    var newTargetPath = Path.Combine(targetPath, identity);
+                    // We need to do this because evaluuated identity in the above line of code can be an empty string
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs]: typo:

> because evaluuated identity

-> because the evaluated identity

**Review by emgarten — APPROVED** [review]
Looks good overall. I suggest testing if using ``/`` causes issues on windows since it is looking for ``\``. It is possible someone might author a project using forward slash because they are also working on linux.

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs (hunk 1 above)
Comment (rohit21agrawal) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs:376]: good catch, added it .

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs (hunk 2 above)
Comment (rohit21agrawal) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs]: fixed

**Comment by rohit21agrawal** [comment]
@emgarten i added tests with both / and \

**Pull request merged**